src/codeu/chat/DerbyStore.java

```diff
@@ -105,36 +106,43 @@ public DerbyStore() {
 	@Override
 	public User getUser(Uuid userid) throws SQLException, IOException {
 		PreparedStatement getUser = conn.prepareStatement(getUserInfo);
-		getUser.setString(1, userid.toString());
+		getUser.setString(1, removeCharsInUuid(userid.toString()));
 		ResultSet user = getUser.executeQuery();
 		
-		return new User(Uuid.parse(user.getString(1)), user.getString(2), Time.fromMs(user.getLong(2)));
+		if (user.next()) {
+			return new User(Uuid.parse(user.getString(1)), user.getString(2), Time.fromMs(user.getLong(4)));
```

Comment:
nit: I'd probably use column names (stored in variables) instead of indexes. Same thing for indexes in other places.